src/index.ts

```diff
@@ -168,5 +173,9 @@ export function valueToEstree(value?: Value): Expression {
       })),
     };
   }
+  if (options && options.jsonFallback) {
+    return valueToEstree(JSON.parse(JSON.stringify(value)));
+  }
```

Comment:
I’m thinking that instead of using JSON serialization, an option `ignorePrototype` can be used. This would then skip the `isPlainObject` check. In other words:

```diff
- if (isPlainObject(value)) {
+ if (options.ignorePrototype || isPlainObject(value)) {
```

This still allows to process values not serializable using JSON on non-plain objects.

Reply:
People can add `toJSON` on their custom objects: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description.
My current method allows that, your method doesn’t.
I do agree that your method is nicer because it “interleaves”, whereas mine is “all or nothing”.
Another alternative is to support that method, perhaps optionally, too?

Reply:
I don’t think `toJSON` should be supported, since this library doesn’t serialize to JSON. It could be added later if a practical use case occurs, but for now, I think the `ignorePrototype` option solves the issue in the cleanest way. (Although I’m open to a better name.)

If `toJSON` should be supported at some point, I think it would be better to call the method instead of actually serializing and parsing it from JSON, as this would allow users to specify a `toJSON` method which returns other types that this function can serialize which aren’t supported by JSON.

Reply:
I’m not sure what you mean exactly :sweat_smile: 

Can we just implement `ignorePrototype` for now and ignore the `toJSON` part? We can revisit it later if this turns out to be needed.

Reply:
> I’m not sure what you mean exactly 😅

I mean exactly what you wrote:

> If toJSON should be supported at some point, I think it would be better to call the method instead of actually serializing and parsing it from JSON, as this would allow users to specify a toJSON method which returns other types that this function can serialize which aren’t supported by JSON.